unison-src/transcripts/idempotent/fix-4536.md

```diff
@@ -0,0 +1,154 @@
+``` ucm
+scratch/main> builtins.merge lib.builtin
+
+  Done.
+```
+
+``` unison :error
+foo : Nat
+foo =
+  use Nat .+
+  1 + 2
+```
+
+``` ucm :added-by-ucm
+  Loading changes detected in scratch.u.
+
+  I got confused here:
+
+      3 |   use Nat .+
+
+
+  I was surprised to find a '.' here.
+  I was expecting one of these instead:
+
+  * bang
+  * binding
+  * do
+  * false
+  * force
+  * handle
+  * if
+  * lambda
+  * let
+  * newline or semicolon
+  * pattern
+  * quote
+  * termLink
+  * true
+  * tuple
+  * typeLink
+```
+
+``` unison :error
+foo : Nat
+foo =
+  use .lib.builtin.Nat +
+  1 + 2
+```
+
+``` ucm :added-by-ucm
+  Loading changes detected in scratch.u.
+
+    😶
+    
+    I was expecting something after the use keyword
+    
+        3 |   use .lib.builtin.Nat +
+    
+    Here's a few examples of valid `use` statements:
+    
+      use math sqrt    Introduces `sqrt` as a local alias for
+                       `math.sqrt`
+      use List :+      Introduces `:+` as a local alias for
+                       `List.:+`.
+      use .foo bar.baz Introduces `bar.baz` as a local alias for
+                       the absolute name `.foo.bar.baz`
```

Comment:
Do we not want to allow absolute names for the first one? The example on line 65 suggests it as correct syntax.

Reply:
I don't see a problem with allowing absolute names in certain places, I can make that change. My instinct was to simplify everything down to relative names, but that could be wrong.